thrall/app/lib/ElasticSearch.scala

```diff
@@ -36,31 +36,33 @@ object ElasticSearch extends ElasticSearchClient with ImageFields {
 
   def currentIsoDateString = printDateTime(new DateTime())
 
-  def indexImage(id: String, image: JsValue)(implicit ex: ExecutionContext): Future[UpdateResponse] = {
-    prepareImageUpdate(id)
+  def indexImage(id: String, image: JsValue)(implicit ex: ExecutionContext): List[Future[UpdateResponse]] = {
+    prepareImageUpdate(id).map( request => request
```

Comment:
```Scala
def setImageCollection(id: String, collections: JsValue)(implicit ex: ExecutionContext): List[Future[ActionResponse]] =
    prepareImageUpdate(id) { request =>
      request.setScriptParams(Map(
        "collections" -> asGroovy(collections),
        "lastModified" -> asGroovy(JsString(currentIsoDateString))
      ).asJava)
        .setScript(
          "ctx._source.collections = collections;" +
            updateLastModifiedScript,
          scriptType)
        .executeAndLog(s"setting collections on image $id")
        .incrementOnFailure(failedCollectionsUpdates) { case e: VersionConflictEngineException => true }
    }

  def prepareImageUpdate(id: String)(op: UpdateRequestBuilder => Future[ActionResponse]) =
    getCurrentIndices.map { list =>
      list.map { index =>
        val updateRequest = client.prepareUpdate(index, imageType, id)
          .setScriptLang("groovy")

        op(updateRequest)
    }}.get
```

That works.
The reason would be that we then don't have to remember to map everytime.